lib/nickserver/adapters/celluloid_http.rb

```diff
@@ -13,7 +13,7 @@ def get(url, options = {})
         params: options[:query],
         ssl_context: ctx,
         ssl_socket_class: Celluloid::IO::SSLSocket
-      yield response.code, response.to_s
+      return response.code, response.to_s
```

Comment:
@elijh this is what i meant above. We were already using fiber based API - but instead of returning the results we kept the blocks on ours side. This resulted in quite a bit of handing down blocks. I think our code is much cleaner in this pseudo synchronous style.